src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/IPrecompiledViewsProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Core.ApplicationParts
+{
+    /// <summary>
+    /// Exposes a sequence of precompiled views associated with an <see cref="ApplicationPart"/> .
+    /// </summary>
+    public interface IPrecompiledViewsProvider
```

Comment:
Can we drop the term `precompiled`? That really only makes sense to us. I think a lot of these APIs would be cleaner and more general without `precompiled` in the names.